witness/src/Dscp/Witness/Web/Types.hs

```diff
@@ -119,35 +160,72 @@ deriveJSON defaultOptions ''BlocksOrMempool
 deriveJSON defaultOptions ''BlockList
 deriveJSON defaultOptions{ omitNothingFields = True } ''BlockInfo
 deriveJSON defaultOptions{ omitNothingFields = True } ''AccountInfo
-deriveJSON defaultOptions{ omitNothingFields = True } ''TxList
-
-instance ToJSON TxInfo where
-    toJSON TxInfo{..} = object $
-        maybe [] (\block -> ["block" .= block]) tiBlock ++
-        case tiTx of
-            GMoneyTx tx ->
-                [ "txId" .= toTxId tx
-                , "txType" .= ("money" :: Text)
-                , "money" .= tx
-                , case sumCoins . map txOutValue . txOuts $ tx of
-                    Right c  -> "txOutValue" .= c
-                    Left err -> "txOutValue" .= err
-                ]
-            GPublicationTx pTx ->
-                [ "txId" .= toPtxId pTx
-                , "txType" .= ("publication" :: Text)
-                , "publication" .= pTx
-                ]
-
-instance FromJSON TxInfo where
-    parseJSON = withObject "tx info" $ \o -> do
+
+instance ToJSON (Detailed a) => ToJSON (WithBlockInfo a) where
+    toJSON WithBlockInfo{..} = mergeObjects
+        (object $ maybe [] (\block -> ["block" .= block]) wbiBlockInfo)
```

Comment:
Why do you need this function `mergeObjects` here, if you just try to add a `block` field to `Detailed a` serialization? Why just not add it manually? Seems like an overkill.

Reply:
What do you mean under "manually"?
I can't prove compiler that `Detailed a` serializes into an object, so can't add a field in a fully safe way.